### ISSUE TITLE
Format Y axis labels

### ISSUE THREAD

**Issue by johannescarlen** [timmolter/XChart#256]
It is possible to change the decimal pattern format of the y-axis, however it would be great if it also was possible to override the whole label formatting. This could also help solving the use of time / date in the y-axis

**Comment by hakantkn** [comment]
You can use custom axis tick calculator. See #217

**Comment by timmolter** [comment]
Yeah, we have `setYAxisDecimalPattern()` but no `setYAxisDatePattern()` like we have for the X-Axis. Why would you want dates in the y-axis though?

You can always use something like this:

```java
      Map<Object, Object> customYAxisTickLabelsMap = new HashMap<>();
      customYAxisTickLabelsMap.put(1.0, "max c");
      customYAxisTickLabelsMap.put(6.0, "max b");
      customYAxisTickLabelsMap.put(9.0, "max a");

      chart.setCustomYAxisTickLabelsMap(customYAxisTickLabelsMap);
```

or

```java
      double[] xData = xySeries.getXData();
      Map<Object, Object> customTickLabelsMap = new HashMap<>();
      SimpleDateFormat sdf = new SimpleDateFormat("MM-dd");

      for (double d : xData) {
        Date date = new Date((long) d);
        String label = sdf.format(date);
        customTickLabelsMap.put(d, label);
      }

      chart.setCustomXAxisTickLabelsMap(customTickLabelsMap);
```

but for the y-axis of course.

**Comment by limitlesslive** [comment]
the new packet version  3.6.1 has no method setYAxisDecimalPattern() ， which version we can use it

**Comment by limitlesslive** [comment]
> Yeah, we have `setYAxisDecimalPattern()` but no `setYAxisDatePattern()` like we have for the X-Axis. Why would you want dates in the y-axis though?
> 
> You can always use something like this:
> 
> ```java
>       Map<Object, Object> customYAxisTickLabelsMap = new HashMap<>();
>       customYAxisTickLabelsMap.put(1.0, "max c");
>       customYAxisTickLabelsMap.put(6.0, "max b");
>       customYAxisTickLabelsMap.put(9.0, "max a");
> 
>       chart.setCustomYAxisTickLabelsMap(customYAxisTickLabelsMap);
> ```
> 
> or
> 
> ```java
>       double[] xData = xySeries.getXData();
>       Map<Object, Object> customTickLabelsMap = new HashMap<>();
>       SimpleDateFormat sdf = new SimpleDateFormat("MM-dd");
> 
>       for (double d : xData) {
>         Date date = new Date((long) d);
>         String label = sdf.format(date);
>         customTickLabelsMap.put(d, label);
>       }
> 
>       chart.setCustomXAxisTickLabelsMap(customTickLabelsMap);
> ```
> 
> but for the y-axis of course.

the new packet version  3.6.1 has no method setYAxisDecimalPattern() ， which version we can use it

**Comment by timmolter** [comment]
This is already supported. Custom Y-axis label formatting — fully replacing the default decimal pattern — is available via:

```java
chart.getStyler().setYAxisTickLabelsFormattingFunction(Function<Double, String>);
// or the convenience shortcut:
chart.setCustomYAxisTickLabelsFormatter(Function<Double, String>);
```

The function receives each raw tick value as a `Double` and returns the exact label string, so any formatting logic works — including the time/date use case you mentioned. Example rendering Y values (epoch millis) as time-of-day:

```java
SimpleDateFormat timeFormat = new SimpleDateFormat("HH:mm");
chart.getStyler()
     .setYAxisTickLabelsFormattingFunction(
         value -> timeFormat.format(new Date(value.longValue())));
```

This produces `01:00`, `01:08`, `01:16`, … on the Y-axis. The symmetric `setXAxisTickLabelsFormattingFunction(...)` exists for the X-axis too.

Added a how-to demo (`TestForIssue256`) in #987. Closing as resolved — please reopen if this doesn't cover your use case.
